develop/api_guc.rst

```diff
@@ -336,30 +339,82 @@ Note that it may be useful to use :code:`error` during development testing, and
   HINT:  Queries are split to multiple tasks if they have to be split into several queries on the workers.
   STATEMENT:  select * from foo;
 
-Real-time executor configuration
-$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
+Adaptive executor configuration
+$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
 
-The Citus query planner first prunes away the shards unrelated to a query and then hands the plan over to the real-time executor. For executing the plan, the real-time executor opens one connection and uses two file descriptors per unpruned shard. If the query hits a high number of shards, then the executor may need to open more connections than max_connections or use more file descriptors than max_files_per_process.
+citus.executor_slow_start_interval (integer)
+********************************************
 
-In such cases, the real-time executor will begin throttling tasks to prevent overwhelming the worker resources. Since this throttling can reduce query performance, the real-time executor will issue an appropriate warning suggesting that increasing these parameters might be required to maintain the desired performance. These parameters are discussed in brief below.
+Time to wait in milliseconds between opening connections to the same worker
+node.
 
-max_connections (integer)
-************************************************
+When the individual tasks of a multi-shard query take very little time, they
+can often be finished over a single (often already cached) connection. To avoid
+redundantly opening additional connections, the executor waits between
+connection attempts for the configured number of milliseconds. At the end of
+the interval, it increases the number of connections it is allowed to open next
+time.
 
-Sets the maximum number of concurrent connections to the database server. The default is typically 100 connections, but might be less if your kernel settings will not support it (as determined during initdb). The real time executor maintains an open connection for each shard to which it sends queries. Increasing this configuration parameter will allow the executor to have more concurrent connections and hence handle more shards in parallel. This parameter has to be changed on the workers as well as the coordinator, and can be done only during server start.
+For long queries (those taking >500ms), slow start simply adds latency, but for
+short queries it's faster. The default value is 10ms.
 
-max_files_per_process (integer)
-*******************************************************
+citus.max_adaptive_executor_pool_size (integer)
```

Comment:
Do you think would it make sense to mention that via `max_adaptive_executor_pool_size `, the users could do `resource management`. For example, if you have some queries that you wouldn't want to utilize too many connections/cores/memory etc., you can lower the value within the session (e.g., this is a session level GUC)